### ISSUE TITLE
All existing jobs from agendaJobs executed after upgrading to pulse 1.6.2

### ISSUE THREAD

**Issue by icelancer** [pulsecron/pulse#49]
### Description

We recently migrated our project from agenda to pulse. However, after updating pulse to version 1.6.2, we encountered an issue where all existing jobs in agendaJobs (accumulated since using agenda) were executed unexpectedly.
This problem did not occur with version 1.6.1.

**Steps to reproduce:**
1. Migrate from agenda to pulse
2. Update pulse from 1.6.1 to 1.6.2
3. Observe that all existing jobs in agendaJobs are executed

**Expected behavior:** Only scheduled jobs should run, not all existing jobs from the previous agenda implementation.
**Actual behavior:** All existing jobs from agendaJobs are executed upon updating to pulse 1.6.2.

**Additional information:**
- The issue was not present in pulse 1.6.1
- This only happened after updating to 1.6.2

**Comment by code-xhyun** [comment]
@icelancer We apologize. We have just become aware of this issue and are working on resolving it. We will address it promptly.

**Comment by icelancer** [comment]
@code-xhyun Thank you for the quick response. I appreciate your attention to this issue and look forward to the resolution.

**Comment by code-xhyun** [comment]
@icelancer Could you provide us with a sample of the data that should not be executed?

**Comment by icelancer** [comment]
Here are two examples of job data that should not have been executed, along with their state after unexpected execution:

**Before execution:**
```
[
  {
    "_id": {"$oid": "63980079d983c26b23c5832d"},
    "data": {
      "type": "sample",
      "resourceId": "df6ed790-fd4c-4b32-b2ea-cdde8992417c"
    },
    "lastFinishedAt": {"$date": "2022-12-13T04:32:57.168Z"},
    "lastModifiedBy": null,
    "lastRunAt": {"$date": "2022-12-13T04:32:57.147Z"},
    "lockedAt": null,
    "name": "update-sample-v2",
    "nextRunAt": null,
    "priority": 0,
    "shouldSaveResult": false,
    "type": "normal"
  },
  {
    "_id": {"$oid": "643567e7c569b650d9f5765a"},
    "data": {
      "type": "sample",
      "resourceId": "df6ed790-fd4c-4b32-b2ea-cdde8992417c"
    },
    "lastFinishedAt": {"$date": "2023-04-11T14:00:07.910Z"},
    "lastModifiedBy": null,
    "lastRunAt": {"$date": "2023-04-11T14:00:07.878Z"},
    "lockedAt": null,
    "name": "update-sample-v2",
    "nextRunAt": null,
    "priority": 0,
    "shouldSaveResult": false,
    "type": "normal"
  }
]
```

**After unexpected execution:**
```
[
  {
    "_id": {"$oid": "63980079d983c26b23c5832d"},
    "data": {
      "type": "sample",
      "resourceId": "df6ed790-fd4c-4b32-b2ea-cdde8992417c"
    },
    "lastFinishedAt": {"$date": "2022-12-13T04:32:57.168Z"},
    "name": "updateGeotag-v2",
    "nextRunAt": {"$date": "2024-08-16T07:56:27.593Z"},
    "priority": 0,
    "shouldSaveResult": false,
    "type": "normal"
  },
  {
    "_id": {"$oid": "643567e7c569b650d9f5765a"},
    "data": {
      "type": "sample",
      "resourceId": "df6ed790-fd4c-4b32-b2ea-cdde8992417c"
    },
    "lastFinishedAt": {"$date": "2023-04-11T14:00:07.910Z"},
    "name": "updateGeotag",
    "nextRunAt": null,
    "priority": 0,
    "shouldSaveResult": false,
    "type": "normal",
    "lastModifiedBy": null,
    "lastRunAt": {"$date": "2024-08-16T07:56:27.652Z"},
    "lockedAt": {"$date": "2024-08-16T07:56:27.642Z"}
  }
]
```

**Comment by code-xhyun** [comment]
@icelancer A version that we believe has solved this issue has been deployed (v 1.6.3).
Please check it out.

[npm package (@latest dist-tag)](https://www.npmjs.com/package/@pulsecron/pulse/v/1.6.3)
[GitHub release](https://github.com/pulsecron/pulse/releases/tag/v1.6.3)

**Comment by icelancer** [comment]
Thank you for your quick response and for addressing this issue promptly. 

I've already updated to version 1.6.3 and can confirm that the problem has been resolved. The unexpected execution of old jobs no longer occurs. 

I appreciate your swift action in fixing this issue and releasing the update. Your responsive support is much appreciated.
